src/newwave/main.mo

```diff
@@ -10,1382 +10,598 @@ import Iter "mo:base/Iter";
+  /**
+   * Public interface for retreiving an entity. Retrieving an entity is subject to any permission and rules defined by the entity.
+   * If the entity is found, the entity is returned, otherwise an error is returned.
+   *
+   * @note Currently there are no permissions and anyone that knows the entity Id can retrieve it
+   * @return The entity if the id matches an entity, otherwise an error
+  */
+  public shared query ({ caller }) func get_entity(entityId : Text) : async Entity.EntityResult {
     let result = getEntity(entityId);
-    return result;
-  };
-
-  public shared ({ caller }) func create_bridge(bridgeToCreate : BridgeEntity.BridgeEntityInitiationObject) : async ?BridgeEntity.BridgeEntity {
-    let result = await createBridge(caller, bridgeToCreate);
-    return result;
-    // return BridgeCreator.create_bridge(bridgeToCreate); TODO: possible to return promise? Would this speed up this canister?
-  };
-
-  public shared query ({ caller }) func get_bridge(entityId : Text) : async ?BridgeEntity.BridgeEntity {
-    let result = getBridge(entityId);
-    return result;
-  };
-
-  public shared query ({ caller }) func get_bridge_ids_by_entity_id(entityId : Text, includeBridgesFromEntity : Bool, includeBridgesToEntity : Bool, includeBridgesPendingForEntity : Bool) : async [Text] {
-    let result = getBridgeIdsByEntityId(entityId, includeBridgesFromEntity, includeBridgesToEntity, includeBridgesPendingForEntity);
-    return result;
+    switch (result) {
+      case (null) { return #Err(#EntityNotFound) };
+      case (?entity) { return #Ok(entity) };
+    };
   };
 
-  public shared query ({ caller }) func get_bridges_by_entity_id(entityId : Text, includeBridgesFromEntity : Bool, includeBridgesToEntity : Bool, includeBridgesPendingForEntity : Bool) : async [BridgeEntity.BridgeEntity] {
-    let result = getBridgesByEntityId(entityId, includeBridgesFromEntity, includeBridgesToEntity, includeBridgesPendingForEntity);
+  /**
+   * Public interface for updating an entity. Only the owner of the entity is allowed to update the values specified in the
+   * entity update object
+   *
+   * @return The entity id if the update was sucessful, otherwise an error
+  */
+  public shared ({ caller }) func update_entity(entityUpdateObject : Entity.EntityUpdateObject) : async Entity.EntityIdResult {
+    let result = await updateEntity(caller, entityUpdateObject);
     return result;
   };
 
-  public shared ({ caller }) func create_entity_and_bridge(entityToCreate : Entity.EntityInitiationObject, bridgeToCreate : BridgeEntity.BridgeEntityInitiationObject) : async (Entity.Entity, ?BridgeEntity.BridgeEntity) {
-    let result = await createEntityAndBridge(caller, entityToCreate, bridgeToCreate);
+  /**
+   * Public interface for deleting an entity. Currently only an owner can delete an entity. This will also delete all the bridges
+   * either pointing to or from this Entity
```

Comment:
TBD how we'll handle this going forward as the Bridges might not be owned be the Entity owner, thus deleting data that belongs to another users. We can go with the deletion for now and come up with a plan how to potentially change this for future iterations.

Reply:
yea agree in general but trying to avoid lingering bridges. To me this seems like a valid case to delete something that the entity owner doesn't own but perhaps we can think of a better way or I guess worse case leave the dangling bridge